### PR TITLE
minitest fixes for negotiate auto and virtual-services

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
Skip tests for unsupported interfaces and inactivate virt svcs.

rubocop: clean
minitest: tested against n30k, n31k, n9k, n9kv

### CODE REVIEW

File: tests/test_interface.rb [hunk 1]
```diff
@@ -559,7 +559,10 @@ def negotiate_auto_helper(interface, default, cmd_ref)
     assert_equal(default, interface.default_negotiate_auto,
                  "Error: #{inf_name} negotiate auto default value mismatch")
 
-    assert_equal(interface.negotiate_auto, default,
+    interface.negotiate_auto = default
```
Comment (mikewiebe) [inline tests/test_interface.rb:562]: Confused by this.. is default the string 'default'? or a method named default

File: tests/test_interface.rb [hunk 2]
```diff
@@ -598,7 +601,7 @@ def negotiate_auto_helper(interface, default, cmd_ref)
     assert_equal(interface.negotiate_auto, non_default,
                  "Error: #{inf_name} negotiate auto value not #{non_default}")
 
-    pattern = cmd_ref.test_config_get_regex[non_default ? 0 : 1]
+    pattern = cmd_ref.test_config_get_regex[non_default ? 1 : 0]
```
Comment (mikewiebe) [inline tests/test_interface.rb:604]: Was this a bug?

Comment (chrisvanheuveln) [inline tests/test_interface.rb:604]: yes

File: tests/test_interface.rb (hunk 1 above)
Comment (chrisvanheuveln) [inline tests/test_interface.rb:562]: It's a string var, passed into the test method.

**Pull request merged**